api/handlers.go

```diff
@@ -159,6 +166,9 @@ func (si *ServerImplementation) verifyHandler(operationID string, ctx echo.Conte
 // LookupAccountByID queries indexer for a given account.
 // (GET /v2/accounts/{account-id})
 func (si *ServerImplementation) LookupAccountByID(ctx echo.Context, accountID string, params generated.LookupAccountByIDParams) error {
+	if params.Round != nil && uint64(*params.Round) > math.MaxInt64 {
+		return badRequest(ctx, errValueExceedingInt64)
+	}
 	if err := si.verifyHandler("LookupAccountByID", ctx); err != nil {
 		return badRequest(ctx, err.Error())
 	}
```

Comment:
Please make sure all of these happen after `verifyHandler`. If the endpoint is disabled we should make sure that is the returned error

Reply:
What do you think about returning `404` errors instead of bad request?

Reply:
I think 404 makes sense. I'll update.